builtin/logical/postgresql/path_role_create.go

```diff
@@ -51,10 +51,15 @@ func (b *backend) pathRoleCreateRead(
 		lease = &configLease{Lease: 1 * time.Hour}
 	}
 
-	// Generate the username, password and expiration
+	// Generate our username, password and expiration.  PostreSQL limits user to
+	// 63 characters.
+	displayName := req.DisplayName
+	if len(displayName) > 40 {
+		displayName = displayName[:40]
+	}
 	username := fmt.Sprintf(
 		"vault-%s-%d-%d",
-		req.DisplayName, time.Now().Unix(), rand.Int31n(10000))
+		displayName, time.Now().Unix(), rand.Int31n(10000))
```

Comment:
Instead of the `timestamp-number` we should use a UUID as it packs entropy in more efficiently. The timestamp is much less secure. We should also do a secondary prune of username to at most 63 characters. Like:

https://github.com/hashicorp/vault/blob/master/builtin/logical/transit/backend_test.go#L18